### ISSUE TITLE
Improve Handling Multiple GatewayClasses

### ISSUE THREAD

**Issue by danehans** [projectcontour/contour#3706]
**Please describe the problem you have**
https://github.com/projectcontour/contour/pull/3659 adds gatewayclass support. The PR handles multiple gatewayclasses by admitting the first one that is reconciled. We should have well defined behavior for handling multiple gatewayclases, i.e. admit the oldest.

xref: https://github.com/projectcontour/contour/pull/3659#discussion_r638178777

**Comment by xaleeks** [comment]
Is this meant to cover the use case of leveraging multiple types of load balancers in a cluster: internal only, external facing, restricted only etc.  Or that they represent different profiles and should be kept these on separate GatewayClasses.  

I have a particular user in mind that requires at least one other load balancer solution in addition to Contour and would like to have this design validated before end of CY Q3.  Can we add this to v1.18?

**Comment by sunjayBhatia** [comment]
As I understand it, Contour will only ever actually reconcile a single GatewayClass. This issue is to make sure if there are multiple with the same controller string/name, Contour correctly sets admitted/not admitted status on them (e.g. the oldest is admitted and any others are set to admitted=false).

**Comment by youngnick** [comment]
Yeah, it's part of the spec that the controller sets the status on things it won't reconcile.

**Comment by sunjayBhatia** [comment]
added to 1.19 for incremental improvements to ensure Operator/Contour interaction works

**Comment by skriss** [comment]
I'll pick this up

**Comment by skriss** [comment]
Working through this, I have the basic logic working but realized that currently, GatewayClass status is being written by all Contour replicas, not just the leader. Need to sort out how to only have the leader pod write status.

**Comment by skriss** [comment]
Got the StatusUpdater wired into the controller to deal with leader election and that seems to be working OK. It looks like there still may be some race conditions/timing issues around startup, so need to do some more digging there.
